apps/gitness/src/App.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import { createBrowserRouter, RouterProvider } from 'react-router-dom'
+import { QueryClientProvider } from '@tanstack/react-query'
+import { ThemeProvider } from '@harnessio/playground'
+import { CodeServiceAPIClient } from '@harnessio/code-service-client'
+import { queryClient } from './framework/queryClient'
+import PipelineList from './pages/pipeline-list'
+import { UI_BASE_URL } from '../app_constants.ts'
+
+export default function App() {
+  React.useEffect(() => {
+    new CodeServiceAPIClient({
+      requestInterceptor: (request: Request): Request => {
+        const apiPrefix = 'api/v1/'
+
+        let newUrl = request.url
+
+        // Check if the request URL starts with the base URL
+        if (request.url.startsWith(UI_BASE_URL)) {
+          // Insert '/api/v1/' after the base URL
+          newUrl = request.url.replace(UI_BASE_URL, `${UI_BASE_URL}${apiPrefix}`)
+        }
+
+        // Create a new Request object with the updated URL and existing options
+        const newRequest = new Request(newUrl, {
```

Comment:
we should use `request.clone()` instead

Reply:
I tried that earlier but looks like `url` can only be updated at the time of creating a request object and `clone` method does not allow changing the url in the cloned instance.

Reply:
Done, as per discussion.